Automattic/wp-calypso
# Introduce splitting CSS by sections

This PR provides an example for how CSS can be split up manually so that we can reduce the size of the initial CSS bundle we load. At a high level this is how I see things being structured:

- JS sections can have an associated SCSS file. This SCSS file will be stored in `/assets/stylesheets/sections`.
- The build script will build each of the SCSS files into a CSS file inside `/public/sections/` and `public/sections-rtl`.
- When we load the page initially, we load the CSS file for that section.
- When we load a new JS chunk, we also load the CSS for that section.

### Comments on this suggestion:
I decided to add a `css` property to the sections that are split out. This allows us to only split some sections for now, and add others in the future.

This solution doesn't deal at all with dependencies. We can split out SCSS files that are inside paths that we know will only be loaded by a section, but all SCSS in components and blocks will need to be loaded in the main bundle. This means that we will still have a very large default bundle.


### Testing instructions:

We should test the following cases:
1. No user boostrapping user logged out
2. No user boostrapping user logged in  ( user need to be set to an LTR and RTL language )
3. User boostrapping enabled && user logged out
4. User boostrapping enabled && user logged in ( user need to be set to an LTR and RTL language )

For each of those we should test a page with a language parameter, such as:
`/log-in/he`
and without language parameter, such as:
`/log-in`

For each of the above we should see that we are loading the correct ( LTR / RTL ) stylesheet and the page is displayed correctly.

With commit https://github.com/Automattic/wp-calypso/pull/17988/commits/b3119cae06fde3634030b62438e286edcba24cd5, we can enable user bootstrapping in `development.json` and set `"wordpress_logged_in_cookie": "username%7your_cookie_here"` in `secrets.json`
  
( this commit should be dropped before merging, we should merge it in https://github.com/Automattic/wp-calypso/pull/18264 )
